### PR TITLE
fix: prompt formatting and method docs in `ChatModel`

### PR DESCRIPTION
**Pull request by NickyHavoc**
# Description
No description.

## Before Merging
 - [x] Review the code changes
    - Unused print / comments / TODOs
    - Missing docstrings for functions that should have them
    - Consistent variable names
    - ...
 - [x] Update `changelog.md` if necessary
 - [x] Commit messages should contain a semantic [label](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) and the ticket number
   - Consider squashing if this is not the case

### CODE REVIEW

File: src/intelligence_layer/core/model.py [hunk 1]
```diff
@@ -533,7 +554,7 @@ class AlephAlphaChatModel(ChatModel, AlephAlphaModel):
     RECOMMENDED_MODELS: ClassVar[list[str]] = []
 
     def to_chat_prompt(
-        self, messages: list[Message], response_prefix: str | None
+        self, messages: list[Message], response_prefix: str | None = None
```
Comment (NiklasKoehneckeAA) [inline src/intelligence_layer/core/model.py:557]: would change the docstring to "Append the given string to the beginning of the final agent message to steer the generation. Defaults to None"

Comment (NiklasKoehneckeAA) [inline src/intelligence_layer/core/model.py:557]: Additionally, you added a default here, but the method that calls this one (generate_chat) does not have a default

Comment (NickyHavoc) [reply src/intelligence_layer/core/model.py:557]: I know, but if we now introduce a default for the others, that would technically be breaking

Comment (NickyHavoc) [reply src/intelligence_layer/core/model.py:557]: Ok, I added the default to only the `to_chat_prompt` function and not to generate/echo chat. This mirrors the behavior of the `ControlModel`, where this parameter is also optional for the `to_instruct_prompt`.

We would have to switch the order of arguments for the `generate_chat` function to give a default here, so I would not do so for now.

**Review by NiklasKoehneckeAA — APPROVED** [review]

**Pull request merged**